.github/workflows/host-tools.yml

```diff
@@ -0,0 +1,111 @@
+
+jobs:
+  build_host_tools:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check latest Haiku hrev
+        id: get_hrev
+        run: |
+          echo "hrev=hrev$(git ls-remote --tags https://review.haiku-os.org/haiku "hrev*" | sed -e 's:.*refs/tags/hrev::g' | sort -nr | head -n1)" >> "$GITHUB_OUTPUT"
+      - name: Check latest host tools release
+        id: get_hosttools
+        run: |
+          hrev=""
+          pageNum=1
+          releaseUrl="${{ github.repository }}/releases/tag/hosttools-${{ env.HAIKU_ARCH }}-"
+          while [ "$hrev" == "" ];
+          do
+            # Fetch using GitHub API
+            json=$(curl -s ${{ github.api_url }}/repos/${{ github.repository }}/releases?page=$pageNum)
+            pageNum=$((pageNum + 1))
+            if [ $(echo $json | jq length) -eq 0 ]; then
+              # This means that we've passed the end and reached an empty array
+              break
+            fi
+            # Store array of revisions
+            revisions=($(echo $json | jq -e -r ".[] | .html_url | select(contains(\"hosttools\"))[${#releaseUrl}:]")) \
+              || continue
+            hrev=${revisions[0]}
+          done
+          echo "hrev=$hrev" >> "$GITHUB_OUTPUT"
+      - name: Build host tools
+        if: steps.get_hrev.outputs.hrev != steps.get_hosttools.outputs.hrev
+        run: |
+          echo "Building host tools at ${{ steps.get_hrev.outputs.hrev }}"
+          hosttools="
+            <build>addattr
+            <build>anyboot
+            <build>bfs_shell
+            <build>collectcatkeys
+            <build>copyattr
+            <build>create_image
+            <build>create_repository_config
+            <build>data_to_source
+            <build>fat_shell
+            <build>fixup_amiga_boot_checksum
+            <build>fixup_tos_boot_checksum
+            <build>fs_shell_command
+            <build>generate_attribute_stores
+            <build>get_package_dependencies
+            <build>linkcatkeys
+            <build>makebootable
+            <build>mbrtool
+            <build>mimeset
+            <build>package
+            <build>package_repo
+            <build>rc
+            <build>resattr
+            <build>rm_attrs
+            <build>set_haiku_revision
+            <build>settype
+            <build>setversion
+            <build>unzip
+            <build>update_package_requires
+            <build>vmdkimage
+            <build>xres
+          "
+          git clone https://review.haiku-os.org/haiku --depth 1
+          git clone https://review.haiku-os.org/buildtools --depth 1
+          sudo apt install -y git nasm bc autoconf automake texinfo flex bison \
+            gawk build-essential unzip wget zip less zlib1g-dev libzstd-dev \
+            xorriso libtool python3 attr
+          cd buildtools/jam
+          make -j$(nproc)
+          cd $GITHUB_WORKSPACE/haiku
+          mkdir generated.${{ env.HAIKU_ARCH }}; cd generated.${{ env.HAIKU_ARCH }}
```

Comment:
Like I mentioned on Matrix, the build folder has to be something that we can assume that's _nearly_ always present on every Linux distro.

See this commit for more details: https://github.com/trungnt2910/haiku-tools/commit/9b9931f8eb069db2f57c5cc8402f19ad5f4b075e

The problematic code can be found [here](https://github.com/haiku/haiku/blob/master/src/build/libroot/fs_attr_generic.cpp#L69), which tries to create a directory at [`HAIKU_BUILD_ATTRIBUTES_DIR`](https://github.com/haiku/haiku/blob/master/src/build/libroot/fs_attr_generic.cpp#L39), which depends on [`$outputDir`](https://github.com/haiku/haiku/blob/master/configure#L538), which is determined using [`$currentDir`](https://github.com/haiku/haiku/blob/master/configure#L532-L536).

I can't say for sure whether it will actually work as I cannot grab the artifacts for this build, see the comment below.

Reply:
Ah okay; kind of weird though, can't say I've seen that problem before. Pushed fix for this.